tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -3576,77 +3604,106 @@ void MSMain(uint GID : SV_GroupIndex,
-    <Shader Name="VS66" Target="vs_6_0" EntryPoint="VSMain" Text="@MAIN" Arguments="/DFALLBACK=1"/>
-    <Shader Name="CS66" Target="cs_6_0" EntryPoint="main" Text="@MAIN" Arguments="/DFALLBACK=1"/>
-    <!- FALLBACK ENDS -->
 
     <Shader Name="MAIN" Target="cs_6_6" EntryPoint="main">
       <![CDATA[
     struct Constants
     {
-      uint idx0, idx1, idx2, idx3, idx4, idx5, idx6, idx7;
+      uint idx0, idx1, idx2, idx3, idx4, idx5, idx6, idx7, idx8, idx9, idx10;
     };
+    
+    // globally declare these so that non-fallback won't throw an unknown identifier error
+    
+    #ifndef NON_UNIFORM
+    #define INDEXER(idx,ix) idx
+    #else
+    #define INDEXER(idx,ix) NonUniformResourceIndex(idx + ix%2)
+    #endif
 
-    #ifdef FALLBACK
 
-    #define ROOT_SIG [RootSignature("RootFlags(ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT), DescriptorTable( SRV(t0), SRV(t1), SRV(t2), UAV(u3), UAV(u4), UAV(u5), CBV(b6), UAV(u7,numDescriptors=3) ), StaticSampler(s0, addressU = TEXTURE_ADDRESS_BORDER, addressV = TEXTURE_ADDRESS_BORDER, borderColor=STATIC_BORDER_COLOR_OPAQUE_WHITE), StaticSampler(s1, addressU = TEXTURE_ADDRESS_BORDER, addressV = TEXTURE_ADDRESS_BORDER, borderColor=STATIC_BORDER_COLOR_OPAQUE_WHITE, comparisonFunc=COMPARISON_EQUAL)")]
+    #ifdef FALLBACK
 
-    ConstantBuffer<Constants> CB : register(b6);
+    #define ROOT_SIG [RootSignature("RootFlags(ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT), DescriptorTable( SRV(t0, numDescriptors=6), UAV(u6, numDescriptors=6), CBV(b12), UAV(u13,numDescriptors=3) ), \
+    DescriptorTable(Sampler(s0, numDescriptors=2), Sampler(s2, numDescriptors=2))")]
+    
+    ByteAddressBuffer g_fallback_rawBuf[2]              : register(t0);
+    StructuredBuffer<float> g_fallback_structBuf[2]     : register(t2);
+    Texture2D<float> g_fallback_tex[2]                  : register(t4);
+    RWByteAddressBuffer g_fallback_rwRawBuf[2]          : register(u6);
+    RWStructuredBuffer<float> g_fallback_rwStructBuf[2] : register(u8);
+    RWTexture1D<float> g_fallback_rwTex[2]              : register(u10);
 
-    ByteAddressBuffer g_rawBuf : register(t0);
-    StructuredBuffer<float> g_structBuf : register(t1);
-    Texture2D<float> g_tex : register(t2);
-    RWByteAddressBuffer g_rwRawBuf : register(u3);
-    RWStructuredBuffer<float> g_rwStructBuf : register(u4);
-    RWTexture1D<float> g_rwTex : register(u5);
+    SamplerState g_fallback_samp[2]              : register(s0);
+    SamplerComparisonState g_fallback_sampCmp[2] : register(s2);
 
+    ConstantBuffer<Constants> CB               : register(b12);
     // Result buffers
-    RWBuffer<float> g_result : register(u7);
-    RWBuffer<float> g_resultVS : register(u8);
-    RWBuffer<float> g_resultPS : register(u9);
-
-    SamplerState g_samp : register(s0);
-    SamplerComparisonState g_sampCmp : register(s1);
-
+    RWBuffer<float> g_result   : register(u13);
+    RWBuffer<float> g_resultVS : register(u14);
+    RWBuffer<float> g_resultPS : register(u15);
```

Comment:
I don't think I can merge the 2 functions into one because Non-Fallback has to access resources via ResourceDescriptorHeap, and fallback needs to access resources using the explicitly bound register array. If Fallback is given ResourceDescriptorHeap, it says "use of undeclared identifier".